### PR TITLE
Add --headers parameter

### LINKED ISSUE #47: Allow passing custom headers and comments for POT file
Currently, `wp i18n make-pot` generates a `.pot` file and adds a link to the WordPress plugin or theme repo page as the `Report-Msgid-Bugs-To` header.

See:

+ https://github.com/wp-cli/i18n-command/blob/master/src/MakePotCommand.php#L375
+ https://github.com/wp-cli/i18n-command/blob/master/src/MakePotCommand.php#L379

For plugins and themes which are not distributed via WordPress repos, adding an optional parameter such as `--bugs_address` could be used to supply an alternate link (e.g. to a GitHub issues page).

### PR DESCRIPTION
**Pull request by greatislander**
This PR resolves #47 by adding a `--headers` parameter which accepts a JSON-formatted array of custom header values which are added to the POT file. If custom headers are passed which correspond to defaults, the default values will be replaced.

### CODE REVIEW

File: src/MakePotCommand.php [hunk 1]
```diff
@@ -163,6 +171,10 @@ public function __invoke( $args, $assoc_args ) {
 			$this->exclude = array_unique( $this->exclude );
 		}
 
+		if ( isset( $assoc_args['headers'] ) ) {
+			$this->headers = json_decode( $assoc_args['headers'], true );
```
Comment (swissspidy) [inline src/MakePotCommand.php]: WP-CLI has a nice little utility method for this so you don't need `isset()` and `json_decode()`.

Example:

```php
$array_arguments = array( 'headers' );
$assoc_args      = \WP_CLI\Utils\parse_shell_arrays( $assoc_args, $array_arguments );
```

File: features/makepot.feature [hunk 2]
```diff
@@ -99,6 +99,24 @@ Feature: Generate a POT file of a WordPress plugin
       "Report-Msgid-Bugs-To: https://wordpress.org/support/plugin/hello-world\n"
       """
 
+  Scenario: Sets custom Report-Msgid-Bugs-To
+    When I run `wp scaffold plugin hello-world`
+
+    When I run `wp i18n make-pot wp-content/plugins/hello-world wp-content/plugins/hello-world/languages/hello-world.pot --headers='{"Report-Msgid-Bugs-To":"https://github.com/hello-world/hello-world/"}'`
+    And the wp-content/plugins/hello-world/languages/hello-world.pot file should contain:
+      """
+      "Report-Msgid-Bugs-To: https://github.com/hello-world/hello-world/\n"
```
Comment (swissspidy) [inline features/makepot.feature]: I would add another test like

```gherkin
And the wp-content/plugins/hello-world/languages/hello-world.pot file should not contain:
  """
  "Report-Msgid-Bugs-To: https://wordpress.org/support/plugin/hello-world\n"
  """
```

to prove that the heaader is indeed being replaced and not added.

File: src/MakePotCommand.php [hunk 3]
```diff
@@ -102,9 +110,9 @@ class MakePotCommand extends WP_CLI_Command {
 	 * @when before_wp_load
 	 */
 	public function __invoke( $args, $assoc_args ) {
-		$this->source  = realpath( $args[0] );
-		$this->slug    = Utils\get_flag_value( $assoc_args, 'slug', Utils\basename( $this->source ) );
-		$this->skip_js = Utils\get_flag_value( $assoc_args, 'skip-js', $this->skip_js );
+		$this->source         = realpath( $args[0] );
+		$this->slug           = Utils\get_flag_value( $assoc_args, 'slug', Utils\basename( $this->source ) );
+		$this->skip_js        = Utils\get_flag_value( $assoc_args, 'skip-js', $this->skip_js );
```
Comment (swissspidy) [inline src/MakePotCommand.php]: Unintended whitespace changes I guess? :-)

**Comment by swissspidy** [comment]
Thanks @greatislander! This is coming along nicely.

**Comment by greatislander** [comment]
@swissspidy Updated based on your feedback -- let me know if there's any other changes you'd like!

**Comment by schlessera** [comment]
Thanks for the pull-request, @greatislander !

**Pull request merged**